guides/dsl.md

```diff
@@ -0,0 +1,241 @@
+# Eve JavaScript/TypeScript DSL
+
+The forthcoming v0.3 release of Eve supports a domain specific langauge (DSL) that interacts with the Eve server, allowing you to intermix Javascript and Eve code. There are several benefits to using the DSL:
+
+1) The syntax is native Javascript, so if you know JS you can write in the DSL
+2) You can use the DSL with Javascript functions and libraries
```

Comment:
sort of, you can't use functions/libraries in the blocks but you can go the other way around if you wanted